plugins/module_utils/common.py

```diff
@@ -0,0 +1,239 @@
+                return False
+            continue
+        if value != dict_two[item]:
+            return False
+    return True
+
+
+def compare_lists(list_one: list, list_two: list) -> bool:
+    """ Compare two unsorted lists """
+    if list_one == list_two:
+        return True
+    # Lists are not sorted and compare is faster than sorting or copying, so...
+    # it might be that I am re-inventing the wheel here...
+    for item in list_one:
+        if item not in list_two:
+            return False
+    for item in list_two:
+        if item not in list_one:
+            return False
+    return True
+
+
+def libvirt_error_to_none(func):
+    """ Decorator to replace libvirt errors to None value return
+
+    When we search for absent object in libvirt we get exception, but
+    sometimes we just want a None value to be returned instead.
+
+    :param func: Callable function to wrap and check the status.
+        """
+
+    def wrapper(self, **kwargs):
+        try:
+            return func(self, **kwargs)
+        except libvirt.libvirtError:
+            return None
+    return wrapper
+
+
+@dataclass
+class ModuleStatus:
+    """ Ansible module status """
+    changed: bool = False
+    failed: bool = False
+    msg: Union[str, None] = None  # linter refuse 'str | None'
+    exception: Union[str, ImportError, None] = None
+    before: Dict = field(default_factory=dict)
+    after: Dict = field(default_factory=dict)
+    warnings: List[str] = field(default_factory=list)
+    data: Dict = field(default_factory=dict)
+
+    @property
+    def report(self):
+        """ Forms a dict suitable for ansible return """
+        ansible_report = {
+            "changed": self.changed,
+            "failed": self.failed,
+            "msg": self.msg
+        }
+        if self.failed and self.exception is not None:
+            ansible_report['exception'] = self.exception
+            return ansible_report
+        if self.changed:
+            # changes may have diff definition, so diff should be added
+            ansible_report['diff'] = {
+                'before': safe_dump(self.before),
+                'after': safe_dump(self.after)}
+            # TODO: Add sort on results to produce better diff
+        # We may have additional data supplied for the return from module
+        if self.data != {}:
+            ansible_report.update(self.data)
+        return ansible_report
+
+
+def data_as_dict(structure):
+    """ Transform dataclass instances to a regular dict """
+    result = []
+    for item in structure:
+        if isinstance(item, list) or isinstance(item, dict) == dict:
+            dict_item = data_as_dict(item)
+        else:
+            dict_item = asdict(item)
+        result.append(dict_item)
+    return result
+
+
+def parse_xml_etree(xml: str) -> dict:
+    """ Parse xml etree and return it as dict """
+    result = {}
+    for element in etree.fromstring(xml).iterchildren():
```

Comment:
is the input here always valid XML, or should it be wrapped in `try`? and do we need to deal with any nested elements?